dot/core/helpers_test.go

```diff
@@ -148,27 +148,11 @@ func createTestService(t *testing.T, genesisFilePath string,
 	return service, encodedExtrinsic
 }
 
-// NewTestService creates a new test core service
-func NewTestService(t *testing.T, cfg *Config) *Service {
+// newTestService creates a new test core service
+func newTestService(t *testing.T, cfg Config) *Service {
 	t.Helper()
 	ctrl := gomock.NewController(t)
 
-	if cfg == nil {
-		cfg = &Config{}
-	}
-
-	if cfg.Keystore == nil {
-		cfg.Keystore = keystore.NewGlobalKeystore()
-		kp, err := sr25519.GenerateKeypair()
-		if err != nil {
-			t.Fatal(err)
-		}
-		err = cfg.Keystore.Acco.Insert(kp)
-		require.NoError(t, err)
-	}
-
-	cfg.LogLvl = 3
```

Comment:
I would rather have this piece of code here once than having this code being repeated everywhere newTestService is used.

That is way too much repetitive. I like the current approach better. In the current approach, `NewTestService` takes values from the config passed to it and for whatever that is not set in config, it would use default config values.

Reply:
I get your point, but when debugging a test this is quite horrible to dig into helping code (and helping code calling other helping code) to find what is configured how. On top of that, often tests don't need all the defaults set and might need only one or two fields set, which makes it a lot easier to understand and modify.

I really pulled my hair changing tests in #2774 which has ramifications in the entire codebase, that's why I made this separate PR.